src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -3410,8 +3412,8 @@ self =>
      * @param isPre specifies whether in early initializer (true) or not (false)
      */
     def templateBody(isPre: Boolean) = inBraces(templateStatSeq(isPre = isPre)) match {
-      case (self, Nil)  => (self, EmptyTree.asList)
-      case result       => result
+      case (selfTypeVal, Nil) => (selfTypeVal, EmptyTree.asList)
+      case result => result
```

Comment:
I think I also tried this as `pair.copy(_2 = EmptyTree.asList)`. I wonder if val looks better, `val pair @ (selfie, stats) = f()`